barretenberg/cpp/src/barretenberg/vm2/simulation/concrete_dbs.hpp

```diff
@@ -45,15 +46,19 @@ class MerkleDB final : public HighLevelMerkleDBInterface {
     MerkleDB(LowLevelMerkleDBInterface& raw_merkle_db,
              PublicDataTreeCheckInterface& public_data_tree_check,
              NullifierTreeCheckInterface& nullifier_tree_check,
-             NoteHashTreeCheckInterface& note_hash_tree_check)
+             NoteHashTreeCheckInterface& note_hash_tree_check,
+             WrittenPublicDataSlotsTreeCheckInterface& written_public_data_slots_tree_check)
         : raw_merkle_db(raw_merkle_db)
         , public_data_tree_check(public_data_tree_check)
         , nullifier_tree_check(nullifier_tree_check)
         , note_hash_tree_check(note_hash_tree_check)
-    {}
+        , written_public_data_slots_tree_check(written_public_data_slots_tree_check)
+    {
+        written_public_data_slots_tree_stack.push(build_public_data_slots_tree());
+    }
 
     // Unconstrained.
-    const TreeSnapshots& get_tree_roots() const override;
+    InternalTreeSnapshots get_tree_roots() const override;
```

Comment:
Hmm, conceptually the same comment as before. I'm not sure this is sound. That is, for "fast simulation" without proving, we wouldn't really want the writes-tracking tree.